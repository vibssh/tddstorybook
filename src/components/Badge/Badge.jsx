import React from 'react';
import PropTypes from 'prop-types';
import './Badge.scss';

const propTypes = {
  children: PropTypes.node,
  type: PropTypes.string
}

const defaultPropTypes = {
  children: null,
  type: 'default'
}

const Badge = (props) => {
  const {children, type} = props;
  const modifier = type === undefined ? '' : type;
  return (
    <div className={`badge ${modifier}`}>{children}</div>
  )
}

Badge.propTypes = propTypes;
Badge.defaultPropTypes = defaultPropTypes;

export default Badge;