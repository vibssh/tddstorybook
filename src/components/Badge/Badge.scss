.badge {
  display: inline-block;
  padding: 0.5em 0.75em;
  border-radius: 20px;
  background: #333;
  color: #fff;
  font-size: 14px;

  &.primary {
    background: #007bff
  }

  &.secondary {
    background: #6c757d;
  }

  &.success {
    background: #28a745;
  }

  &.danger {
    background: #dc3545;
  }

  &.warning {
    background: #ffc107;
  }

  &.info {
    background: #17a2b8;
  }

  &.light {
    background: #f8f9fa;
    color: #333;
  }

  &.dark {
    background: #343a40;
    color: #fff;
  }
}