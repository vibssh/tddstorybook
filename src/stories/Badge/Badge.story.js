import React from "react";
import { storiesOf } from "@storybook/react";

import Badge from "../../components/Badge";

const type = {
  DEFAULT: "default",
  PRIMARY: "primary",
  SECONDARY: "secondary",
  SUCCESS: "success",
  DANGER: "danger",
  WARNING: "warning",
  INFO: "info",
  LIGHT: "light",
  DARK: "dark"
};

const iterator = () => {
  const Types = Object.keys(type);
  Types.map(T =>
    storiesOf("Badge", module).addWithJSX(`Badge ${type[T]}`, () => (
      <Badge type={type[T]}>{`${type[T]}`}</Badge>
    ))
  );
};

iterator();
