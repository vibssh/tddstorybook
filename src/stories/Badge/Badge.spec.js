import React from 'react';
import {shallow} from 'enzyme';

import Badge from '../../components/Badge';

const Types = {
  DEFAULT: "default",
  PRIMARY: "primary",
  SECONDARY: "secondary",
  SUCCESS: "success",
  DANGER: "danger",
  WARNING: "warning",
  LIGHT: "light"
};

describe('Badge', () => {
  let wrapper;
  describe('should render default badge', () => {
    it('should have a default badge ', () => {
      wrapper = shallow(
      <Badge>Default Badge</Badge>
      )
      expect(wrapper.debug()).toMatchSnapshot()
    })
  })
  describe('should render primary badge', () => {
    it('should have a primary badge', () => {
      wrapper = shallow(
        <Badge type={Types['PRIMARY']}>Primary Badge</Badge>
      )
      expect(wrapper.debug()).toMatchSnapshot();
    })
  })
})