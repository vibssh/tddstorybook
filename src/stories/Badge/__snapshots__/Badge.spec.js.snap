// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render default badge should have a default badge  1`] = `
"<div className=\\"badge \\">
  Default Badge
</div>"
`;

exports[`Badge should render primary badge should have a primary badge 1`] = `
"<div className=\\"badge primary\\">
  Primary Badge
</div>"
`;
